prog/vm/nexus.rb

```diff
@@ -197,11 +198,29 @@ def create_storage_volume_records
         boot: volume["boot"],
         size_gib: volume["size_gib"],
         disk_index: disk_index,
-        key_encryption_key_1_id: key_encryption_key&.id
+        key_encryption_key_1_id: key_encryption_key&.id,
+        spdk_installation_id: allocate_spdk_installation(vm_host.spdk_installations)
       )
     end
   end
 
+  def allocate_spdk_installation(spdk_installations)
+    # YYY: remove after upgrading all legacy servers
+    return nil if spdk_installations.empty?
+
+    total_weight = spdk_installations.sum(&:allocation_weight)
+    fail "Total weight of all spdk_installations shouldn't be zero." if total_weight == 0
+
+    rand_point = rand(0..total_weight - 1)
+    weight_sum = 0
+    rand_choice = spdk_installations.each { |si|
+      break si if weight_sum + si.allocation_weight > rand_point
+      weight_sum += si.allocation_weight
```

Comment:
nit: if you do addition first, you don't need to do addition second time, you can check against weight_sum/